src/StreamingBodyParser/ParserInterface.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace React\Http\StreamingBodyParser;
+
+use Evenement\EventEmitterInterface;
+use React\Http\Request;
+
+interface ParserInterface extends EventEmitterInterface
+{
+    public function __construct(Request $request);
```

Comment:
You have Factory method, so this antipattern `__construct` don't need to be in interface.